### PR TITLE
[web-animations-1] Add parameter for fetching all animations on a subtree

### LINKED ISSUE #2058: [web-animations-1] Add parameter to getAnimations for getting animations in descendants too
_From @birtles on March 17, 2016 0:48_

As discussed and minuted here: The ideas we came up with are here: https://docs.google.com/document/d/1lQbgJ0fZYGn1_qboj59G-MOILK33z4kmBdjgtSHF2gA/edit#heading=h.cioj28lsgoa2

A more concrete elaboration on that proposal is described here:
https://bugzilla.mozilla.org/show_bug.cgi?id=1254761#c0

_Copied from original issue: w3c/web-animations#146_

### PR DESCRIPTION
**Pull request by birtles**
This fixes #2058.

### CODE REVIEW

File: web-animations-1/Overview.bs [hunk 1]
```diff
@@ -5721,11 +5726,14 @@ animation.play();</pre>
...
-    sequence&lt;Animation&gt; getAnimations()</dfn>
+:   <dfn method for=Animatable lt="getAnimations(options)">
+    sequence&lt;Animation&gt; getAnimations(options)</dfn>
 ::  Returns the set of [=relevant=] {{Animation}} objects
     that contain at least one <a>animation effect</a> whose
-    <a>target element</a> is this object.
+    <a>target element</a> is this object, or,
+    if an {{Animatable/getAnimations(options)/options}} parameter
+    is passed with {{GetAnimationsOptions/subtree}} set to true,
+    an [=inclusive descendant=] of this object.
```
Comment (stephenmcgruer) [inline web-animations-1/Overview.bs]: Do we need to specify anything regarding shadow DOM, or does inclusive descendant automatically exclude them?

Comment (birtles) [reply web-animations-1/Overview.bs]: As I understand it, "inclusive descendant" automatically excludes them. To include them you need to use the [shadow including inclusive descendant](https://dom.spec.whatwg.org/#concept-shadow-including-inclusive-descendant) concept.

Comment (stephenmcgruer) [reply web-animations-1/Overview.bs]: Ack, thanks.

**Comment by annevk** [comment]
So the idea is to not tackle shadow roots at all? Will you add tests to ensure nothing from shadow roots is returned?

**Comment by birtles** [comment]
> So the idea is to not tackle shadow roots at all? Will you add tests to ensure nothing from shadow roots is returned?

Right. Yes, I need to update the implementation in Firefox to match the spec terms and I plan to add WPT at that point. I'll make sure to add one for shadow roots at the same time.

**Pull request merged**